lib/marginalia/comment.rb

```diff
@@ -4,11 +4,16 @@ module Marginalia
   module Comment
     mattr_accessor :components, :lines_to_ignore
     Marginalia::Comment.components ||= [:application, :controller, :action]
+    THREAD_LOCAL_KEY_FOR_JOB = 'marginalia.comment.job'
 
     def self.update!(controller = nil)
       @controller = controller
     end
 
+    def self.update_job!(job)
+      Thread.current[THREAD_LOCAL_KEY_FOR_JOB] = job
```

Comment:
Actually it looks like that problem was solved in https://github.com/basecamp/marginalia/commit/fc51f4443b5286d4b239c45b4f6be4f6e442be08. I'll rebase and update this PR soon to make the style more consistent with that commit (e.g., using a Symbol for the thread local key)